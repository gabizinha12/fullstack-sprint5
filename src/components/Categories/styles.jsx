import styled from "styled-components";

export const Navbar = styled.nav`
  @media (max-width: 1200px) {
    display: none;
  }
`;

export const UnorderedList = styled.ul`
  display: flex;
  justify-content: space-between;
  margin-top: 14px;
  width: 100%;
  margin-bottom: 14px;
`;
