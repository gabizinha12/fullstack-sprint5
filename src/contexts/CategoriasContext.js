import React from "react";

export const CategoriasContext = React.createContext();

const INITAL_STATE = {
  all: [],
  current: [],
};

export const CatgoriasContext = ({ children }) => {
  const [categorias, setCategorias] = React.useState(INITAL_STATE);

  return (
    <CategoriasContext.Provider value={{ categorias, setCategorias }}>
      {children}
    </CategoriasContext.Provider>
  );
};
