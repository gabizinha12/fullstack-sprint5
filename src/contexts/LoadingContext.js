import React, { createContext, useState } from "react";
export const LoadingContext = createContext();
const INITIAL_STATE = {
  show: false,
};
export const LoadContext = ({ children }) => {
  const [loading, setLoading] = useState(INITIAL_STATE);
  return (
    <LoadingContext.Provider value={{ loading, setLoading }}>
      {children}
    </LoadingContext.Provider>
  );
};
