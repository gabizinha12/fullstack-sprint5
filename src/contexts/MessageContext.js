import React, { createContext, useState } from "react";

export const MessageContext = createContext();

export function MessageProvider({ children }) {
  const [hasError, setHasError] = useState(false);

  return (
    <MessageContext.Provider value={{ hasError, setHasError }}>
      {children}
    </MessageContext.Provider>
  );
}
