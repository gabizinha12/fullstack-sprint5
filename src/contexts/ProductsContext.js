import React from "react";

export const ProductsContext = React.createContext();
const initial_state = {
  products: [],
  filters: [],
};
export const ProdutosContext = ({ children }) => {
  const [produtos, setProdutos] = React.useState(initial_state);
  return (
    <ProductsContext.Provider value={{ produtos, setProdutos }}>
      {children}
    </ProductsContext.Provider>
  );
};
