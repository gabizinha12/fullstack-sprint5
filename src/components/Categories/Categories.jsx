import { CategoriasContext } from "../../contexts/CategoriasContext";
import { CategoriesItem } from "./CategoriesItem/CategoriesItem";
import { Navbar, UnorderedList } from "./styles";

export function Categories() {
  const categorias = useContext(CategoriasContext);

  return (
    <Navbar>
      <UnorderedList>
        {categorias.categorias.all.map((el) => (
          <CategoriesItem key={el.id} label={el.label} link={el.link} />
        ))}
      </UnorderedList>
    </Navbar>
  );
}
