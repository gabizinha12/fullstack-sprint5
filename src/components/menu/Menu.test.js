import { render, screen } from "@testing-library/react";
import Menu from "./Menu";
import { Categories } from "../Categories/Categories";
import { CategoriasContext } from "../../contexts/CategoriasContext";
describe("Teste de snapshot Menu", () => {
  it("should render Menu", () => {
    const setCategorias = jest.fn();
    const categorias = null;
    const { container } = render(
      <Menu>
        <CategoriasContext.Provider value={{ categorias, setCategorias }}>
          <Categories />
        </CategoriasContext.Provider>
      </Menu>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
