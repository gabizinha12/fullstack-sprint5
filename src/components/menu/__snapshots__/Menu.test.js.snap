// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de snapshot Menu should render Menu 1`] = `
<div
  class="sc-gtsrHT cfEjjb"
/>
`;
